web/pdf_presentation_mode.js

```diff
@@ -119,17 +119,34 @@ var PDFPresentationMode = (function PDFPresentationModeClosure() {
       return true;
     },
 
-    /**
-     * Switches page when the user scrolls (using a scroll wheel or a touchpad)
-     * with large enough motion, to prevent accidental page switches.
-     * @param {number} delta - The delta value from the mouse event.
-     */
-    mouseScroll: function PDFPresentationMode_mouseScroll(delta) {
+    _mouseWheel: function PDFPresentationMode_mouseWheel(evt) {
       if (!this.active) {
         return;
       }
+
+      evt.preventDefault();
+
+      var delta = Math.sqrt(evt.deltaX * evt.deltaX + evt.deltaY * evt.deltaY);
+      var angle = Math.atan2(evt.deltaY, evt.deltaX);
+      if (-0.25 * Math.PI < angle && angle < 0.75 * Math.PI) {
+        // All that is left-up oriented has to change the sign.
+        delta = -delta;
+      }
+
+      var MOUSE_DOM_DELTA_PIXEL_MODE = 0;
+      var MOUSE_DOM_DELTA_LINE_MODE = 1;
+      var MOUSE_DOM_DELTA_PAGE_MODE = 2;
+      var MOUSE_PIXELS_PER_LINE = 30;
+      var MOUSE_LINES_PER_PAGE = 30;
+      if (evt.deltaMode === MOUSE_DOM_DELTA_PIXEL_MODE) {
+        delta /= MOUSE_PIXELS_PER_LINE * MOUSE_LINES_PER_PAGE;
+      } else if (evt.deltaMode === MOUSE_DOM_DELTA_LINE_MODE) {
+        delta /= MOUSE_LINES_PER_PAGE;
+      }
+      // The delta is in per-page units.
```

Comment:
Oh, probably. In that case, it might be better to change it to `The delta is now in per-page units` to indicate this it's now the case, but not before.